csharp-basic-topics/RemoveIllegalCharsFromPathAndFilenames/RemoveIllegalCharsFromPathAndFilenames/Program.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Text.RegularExpressions;
+
+namespace CodeMazeGuides.CSBasic
+{
+    public class Program
+    {
+        static void Main(string[] args)
+        {
+            List<string> pathList = new List<string> 
+            {
+                "C://Thisisavalidpath",
+                "D://Thi*IsINVALID/<<",
+                "E://Users/ValidPath/docs",
+                "F://Users/Photos/Cats",
+                "G://Inv@7***lidPath",
+                "H://Test/This<IS>Ano<>therInvalidPath"
+            };
```

Comment:
Just to note if running this in a Unix/Linux environment, all of the above paths have invalid filename characters, and so the code that combines both invalid path and invalid file name chars is going to return this entire list.

We should be sure within the context of the article and our code that we are not "windows specific", since .NET is cross platform and there are overrides of the GetInvalidPath methods that are platform specific within the framework (just have a look at the runtime source code). If we want, in the code we could if block around the specific OSs and provide OS specific examples, which would be cool.

Reply:
How would you try to do that? Would you create two different projects one targeting windows framework, the other one linux? Same for tests? Or would you add preprocessor directives in the projects? For different input data.